### PR TITLE
Add scripts for deploying Hyperlane

### PR DESCRIPTION
**Pull request by asaj**
### Description

This PR initializes the repo with scripts for deploying Hyperlane core contracts, MultisigIsms, and sending test messages.

### Drive-by changes

- CI and other repo initialization tasks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,78 @@
...
+- A `ProxyAdmin`, used to administer `TransparentUpgradableProxies` for `Mailbox` and `InterchainGasPaymaster`
+- A `Mailbox`, which applications can use to send and receive messages
+- A `MultisigIsm`. Applications can optionally use this ISM to verify interchain messages sent to the local chain.
+- An `InterchainGasPaymaster`. Applications can optionally use this contract to pay a relayer to deliver their interchain messages to remote chains.
+- A `TestRecipient`. Users can send messages to this contract to verify that everything is working properly.
+
+```bash
+# The private key that will be used to deploy the contracts. Does not have any
+# permissions post-deployment, any key with a balance will do.
+export PRIVATE_KEY=0x1234
```
Comment (nambrot) [inline README.md:45]: I think we can avoid needing to prescribe the use of `PRIVATE_KEY` env vars and instead be unopinionated around the tx signature method by leveraging the native wallet options of foundry?

File: config/networks.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  "test1": {
+    "id": 13371
+  },
+  "test2": {
+    "id": 13372
+  },
+  "test3": {
+    "id": 13373
+  }
+}
```
Comment (nambrot) [inline config/networks.json]: Nit: Newline
```suggestion
}

```

File: lib/CheckLib.sol [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import {ConfigLib} from "../lib/ConfigLib.sol";
+import {MultisigIsm} from "@hyperlane-xyz/core/contracts/isms/MultisigIsm.sol";
+import {TransparentUpgradeableProxy} from "@hyperlane-xyz/core/contracts/upgrade/TransparentUpgradeableProxy.sol";
+
+library CheckLib {
+    function check(
+        ConfigLib.HyperlaneDomainConfig memory config,
+        ConfigLib.MultisigIsmConfig memory ismConfig
+    ) internal view {
+        // checkBytecode(config);
```
Comment (nambrot) [inline lib/CheckLib.sol]: Remove?

File: scripts/DeployCore.s.sol [hunk 4]
```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import "../lib/forge-std/src/Script.sol";
+
+import {ConfigLib} from "../lib/ConfigLib.sol";
+import {CheckLib} from "../lib/CheckLib.sol";
+import {DeployLib} from "../lib/DeployLib.sol";
+
+// TODO: Verification
```
Comment (nambrot) [inline scripts/DeployCore.s.sol]: Optional Nit: Can we file issues for some of these TODOs?

File: scripts/SendTestMessage.s.sol [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+import "../lib/forge-std/src/Script.sol";
+import "../lib/forge-std/src/console.sol";
+
+import {BytesLib} from "../lib/BytesLib.sol";
+import {ConfigLib} from "../lib/ConfigLib.sol";
+import {Mailbox} from "@hyperlane-xyz/core/contracts/Mailbox.sol";
+import {TypeCasts} from "@hyperlane-xyz/core/contracts/libs/TypeCasts.sol";
+
+// TODO: Maybe take recipient as an arg..
```
Comment (nambrot) [inline scripts/SendTestMessage.s.sol]: What exactly does this mean here?

**Review by nambrot — APPROVED** [review]
Very impressive and cool!

File: scripts/SendTestMessage.s.sol (hunk 5 above)
Comment (asaj) [reply scripts/SendTestMessage.s.sol]: Outdated

**Pull request merged**